build.sh

```diff
@@ -74,6 +65,59 @@ fetch_krun() {
     fi
 }
 
+# We build our own fixed version of GCC, thus ruling out differences in
+# packaged compilers for the different platforms.
+GCC_V=4.9.3
+OUR_CC=${wrkdir}/gcc-inst/bin/zgcc
+OUR_CXX=${wrkdir}/gcc-inst/bin/zg++
+GCC_TARBALL_URL=ftp://ftp.mirrorservice.org/sites/ftp.gnu.org/gnu/gcc/gcc-${GCC_V}/gcc-${GCC_V}.tar.gz
+build_gcc() {
+    echo "\n===> Download and build GCC\n"
+    if [ -f ${OUR_CC} ]; then return; fi
+    cd ${wrkdir}
+    if ! [ -f ${wrkdir}/gcc-${GCC_V}.tar.gz ]; then
+        wget ${GCC_TARBALL_URL} || exit $?
+    fi
+    if ! [ -d ${wrkdir}/gcc ]; then
+        tar xfzp gcc-${GCC_V}.tar.gz || exit $?;
+        mv gcc-${GCC_V} gcc || exit $?
+    fi
+    cd gcc || exit $?
+
+    # OpenBSD needs patches
+    if [ `uname` -eq "openbsd" ]; then
+        for i in ../../patches/openbsd_gcc_patches/patch*; do
+            patch -Ep0 < $i || exit $?
+        done
+    fi
+
+    # download script uses fixed versions, so OK.
+    ./contrib/download_prerequisites || exit $?
+
+    mkdir sd_build || exit $?
+    cd sd_build || exit $?
+
+    ../configure \
+        --prefix=${wrkdir}/gcc-inst \
+        --disable-libcilkrts \
+        --program-transform-name=s,^,z, \
+        --verbose \
+        --disable-libmudflap \
+        --disable-libgomp \
+        --disable-tls \
+        --enable-languages=c,c++ \
+        --with-system-zlib \
+        --disable-tls \
+        --enable-threads=posix \
+        --enable-wchar_t \
+        --disable-libstdcxx-pch \
+        --enable-cpp \
+        --enable-shared \
+      || exit $?
+    $MYMAKE -j4 || exit $?
```

Comment:
I wonder if we should be less aggressive with `-j`? If machines have low-ish RAM, we might accidentally force machines into swap?